Sources/MapboxCoreNavigation/FixLocation.swift

```diff
@@ -10,7 +10,7 @@ extension FixLocation {
         }
         
         self.init(coordinate: location.coordinate,
-                  monotonicTimestampNanoseconds: 0, // use time instead; see also `Navigator.status(at:)`
+                  monotonicTimestampNanoseconds: Int64(location.timestamp.nanosecondsSince1970),
```

Comment:
NavNative explicitly requested this change.

Reply:
Does the navigator expect `monotonicTimestampNanoseconds` to be monotonic? It won’t be in some cases like when moving across time zone boundaries. As I understand it, MapboxNavigationNative added support for monotonic timestamps for Android location providers in particular, which are even more likely to have their normal timestamps go back and forth.

A monotonic timestamp would come from the system, not from a particular location. However, on iOS, that monotonic timestamp would be less informative than a normal location timestamp. It’s much, much more common for Core Location to give us multiple batched location updates with the same timestamp. It’s no problem if the navigator is using `monotonicTimestampNanoseconds` as essentially an alias for `time` (but converted to nanoseconds), but we should make sure MapboxNavigationNative isn’t making any assumptions about `monotonicTimestampNanoseconds` that don’t hold with regard to `CLLocation.timestamp`.

/cc @SiarheiFedartsou @kmadsen

Reply:
> Does the navigator expect monotonicTimestampNanoseconds to be monotonic? 

It is desirable, but not required. The only thing that we shouldn't have significant "jumps" backwards in time.

>  As I understand it, MapboxNavigationNative added support for monotonic timestamps for Android location providers in particular, which are even more likely to have their normal timestamps go back and forth.

On Android they have special field in Location which is based on system uptime and is guaranteed to be monotonic.

> but we should make sure MapboxNavigationNative isn’t making any assumptions about monotonicTimestampNanoseconds that don’t hold with regard to CLLocation.timestamp.

Actually at the moment if `monotonicTimestampNanoseconds` is zero we simply convert `time` field to it and use "as is". This change was required, because some new parts of NN rely on the fact that you fill this `monotonicTimestampNanoseconds` field. Actually we consider `time` as just additional field which could help in debug for example, but `monotonicTimestampNanoseconds` should be used instead.